diff --git a/kernel/events/core.c b/kernel/events/core.c
index d4037a6ead73..8ecda297e48f 100644
--- a/kernel/events/core.c
+++ b/kernel/events/core.c
@@ -4870,7 +4870,7 @@ static struct perf_event_pmu_context *
 find_get_pmu_context(struct pmu *pmu, struct perf_event_context *ctx,
 		     struct perf_event *event)
 {
-	struct perf_event_pmu_context *new = NULL, *pos = NULL, *epc;
+	struct perf_event_pmu_context *new = NULL, *epc;
 	void *task_ctx_data = NULL;
 
 	if (!ctx->task) {
@@ -4927,19 +4927,12 @@ find_get_pmu_context(struct pmu *pmu, struct perf_event_context *ctx,
 			atomic_inc(&epc->refcount);
 			goto found_epc;
 		}
-		/* Make sure the pmu_ctx_list is sorted by PMU type: */
-		if (!pos && epc->pmu->type > pmu->type)
-			pos = epc;
 	}
 
 	epc = new;
 	new = NULL;
 
-	if (!pos)
-		list_add_tail(&epc->pmu_ctx_entry, &ctx->pmu_ctx_list);
-	else
-		list_add(&epc->pmu_ctx_entry, pos->pmu_ctx_entry.prev);
-
+	list_add(&epc->pmu_ctx_entry, &ctx->pmu_ctx_list);
 	epc->ctx = ctx;
 
 found_epc:
@@ -5863,15 +5856,14 @@ static int _perf_event_period(struct perf_event *event, u64 value)
 	if (!value)
 		return -EINVAL;
 
-	if (event->attr.freq) {
-		if (value > sysctl_perf_event_sample_rate)
-			return -EINVAL;
-	} else {
-		if (perf_event_check_period(event, value))
-			return -EINVAL;
-		if (value & (1ULL << 63))
-			return -EINVAL;
-	}
+	if (event->attr.freq && value > sysctl_perf_event_sample_rate)
+		return -EINVAL;
+
+	if (perf_event_check_period(event, value))
+		return -EINVAL;
+
+	if (!event->attr.freq && (value & (1ULL << 63)))
+		return -EINVAL;
 
 	event_function_call(event, __perf_event_period, &value);
 
@@ -8117,8 +8109,7 @@ void perf_event_exec(void)
 
 	perf_event_enable_on_exec(ctx);
 	perf_event_remove_on_exec(ctx);
-	scoped_guard(rcu)
-		perf_iterate_ctx(ctx, perf_event_addr_filters_exec, NULL, true);
+	perf_iterate_ctx(ctx, perf_event_addr_filters_exec, NULL, true);
 
 	perf_unpin_context(ctx);
 	put_ctx(ctx);
